ecs-init/docker/docker.go

```diff
@@ -440,6 +463,33 @@ func (c *client) getHostConfig(envVarsFromFiles map[string]string) *godocker.Hos
 	return createHostConfig(binds)
 }
 
+// getCredentialsFetcherSocketBind returns the bind for CredentialsFetcher socket and also whether the bind exists.
+// Value for the bind is as follow:
+// CREDENTIALS_FETCHER_HOST (as in os.Getenv) set: source CREDENTIALS_FETCHER_HOST (as in os.Getenv, trim unix:// prefix),
+// dest CREDENTIALS_FETCHER_HOST (as in /etc/ecs/ecs.config, trim unix:// prefix)
+func getCredentialsFetcherSocketBind(envVarsFromFiles map[string]string) (string, bool) {
+	CredentialsFetcherEndpointAgent := defaultCredentialsFetcherSocketPath
+	CredentialsFetcherUnixSocketSourcePath, fromEnv := config.CredentialsFetcherUnixSocket()
+	if fromEnv {
```

Comment:
why are we doing special parsing only in the case where the host socket path comes from env var?

Reply:
yes, this it if the variable comes from env